### PR TITLE
imp: balance: Allow --related flag for balance reports. (#1469)

### LINKED ISSUE #1469: Display related accounts when generating cash flow report
GnuCash's Cash Flow Report lists related accounts for a given set of selected cash accounts for both "Money into" and "Money out of" sections. Adapted from an IRC conversation with @simonmichael, one way of emulating this is the command

```
hledger print CASHACCTSQUERY | hledger -f- -I bal --invert not:CASHACCTSQUERY
```

Could we add support for the `-r`/`--related` flag to the balance commands and cashflow report?

### PR DESCRIPTION
**Pull request by Xitian9**
Partially fixes #1469.

### CODE REVIEW

File: hledger/test/related.test [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# 1. With --related, balance will on related accounts
```
Comment (simonmichael) [inline hledger/test/related.test:1]: typo ("on")

**Review by simonmichael — COMMENTED** [review]
I don't have a clear sense of where I'd use this, but it seems harmless and consistent. I think we just need to mention it in Balance.md.

**Comment by Xitian9** [comment]
Documentation added.

**Comment by Xitian9** [comment]
Anything more needed for this?

**Comment by simonmichael** [comment]
No, thanks!

**Comment by simonmichael** [comment]
a

**Pull request merged**